src/utility.rs

```diff
@@ -114,163 +200,111 @@ pub(crate) fn get_coverage_perc(covs: &[Value]) -> Result<f64, SifisError> {
+        || sifis_quantized > thresholds[1]
+        || crap > thresholds[2]
+        || skunk > thresholds[3]
+}
+
+// Get AVG MIN MAx and the list of all complex files
+pub(crate) fn get_cumulative_values(
+    metrics: &Vec<Metrics>,
+) -> (Metrics, Metrics, Metrics, Vec<Metrics>) {
+    let mut avg = Metrics::default();
+    let mut min = Metrics::default();
+    let mut max = Metrics::default();
+    avg.file = "AVG".to_string();
+    max.file = "MAX".to_string();
+    min.file = "MIN".to_string();
+    let mut complex_files = Vec::<Metrics>::new();
+    metrics.iter().for_each(|m| {
         avg.sifis_plain += m.sifis_plain;
         avg.crap += m.crap;
         avg.skunk += m.skunk;
         avg.sifis_quantized += m.sifis_quantized;
-        min.sifis_plain = min.sifis_plain.min(m.sifis_plain);
-        min.crap = min.crap.min(m.crap);
-        min.sifis_quantized = min.sifis_quantized.min(m.sifis_quantized);
-        min.skunk = min.skunk.min(m.skunk);
+        avg.coverage += m.coverage * 100.;
         max.sifis_plain = max.sifis_plain.max(m.sifis_plain);
-        max.crap = max.crap.max(m.crap);
         max.sifis_quantized = max.sifis_quantized.max(m.sifis_quantized);
+        max.crap = max.crap.max(m.crap);
         max.skunk = max.skunk.max(m.skunk);
-    }
+        min.sifis_plain = min.sifis_plain.min(m.sifis_plain);
+        min.sifis_quantized = min.sifis_quantized.min(m.sifis_quantized);
+        min.crap = min.crap.min(m.crap);
+        min.skunk = min.skunk.min(m.skunk);
+        if m.is_complex {
+            complex_files.push(m.clone());
+        }
+    });
     avg.sifis_plain /= metrics.len() as f64;
     avg.crap /= metrics.len() as f64;
     avg.skunk /= metrics.len() as f64;
     avg.sifis_quantized /= metrics.len() as f64;
-    (avg, min, max)
+    avg.coverage /= metrics.len() as f64;
+    (avg, max, min, complex_files)
 }
 
-pub(crate) fn export_to_json(
-    project_folder: &Path,
-    output_path: &Path,
-    metrics: Vec<Metrics>,
-    files_ignored: Vec<String>,
-) -> Result<(), SifisError> {
-    let n_files = files_ignored.len();
-    let json = json!({
-        "project": project_folder.display().to_string(),
-        "numver_of_files_ignored": n_files,
-        "metrics":metrics,
-        "files_ignored":files_ignored,
-    });
-    let json_string = match serde_json::to_string(&json) {
-        Ok(data) => data,
-        Err(_err) => return Err(SifisError::WrintingError()),
-    };
-    match fs::write(output_path, json_string) {
-        Ok(_ok) => (),
-        Err(_err) => return Err(SifisError::WrintingError()),
-    };
-    Ok(())
+// Calculate SIFIS PLAIN , SIFIS QUANTIZED, CRA and SKUNKSCORE for the entire project
+// using the sum values computed before
+pub(crate) fn get_project_metrics(
+    project_coverage: f64,
+    sifis_plain_sum: f64,
+    sifis_quantized_sum: f64,
+    ploc_sum: f64,
+    comp_sum: f64,
+) -> Metrics {
+    let complexity_factor = 25.0;
+    Metrics {
```

Comment:
Use the new `pub struct Config` for parameters and then a `new` function to pass the `Config` struct

Reply:
Done, now this function use the Config struct